code/game/objects/structures/grey_autocloner.dm

```diff
@@ -0,0 +1,151 @@
+	var/datum/gas_mixture/air_contents = new()
+	var/list/brine_types = list("corazone", "perfluorodecalin", "epinephrine", "salglu_solution") //Taken from cloner, a bit of extra healing though they should be fully good.
+	var/datum/mind/clonemind
+	var/attempting = FALSE
+
+/obj/machinery/grey_autocloner/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/implanter))
+		var/obj/item/implanter/I = O
+		if(istype(I.imp, /obj/item/implant/grey_autocloner))
+			var/obj/item/implant/grey_autocloner/G = I.imp
+			G.linked = src
+			atom_say("Link confirmed!")
+	return ..()
+
+/obj/machinery/grey_autocloner/Initialize(mapload)
+	. = ..()
+
+	air_contents.oxygen = MOLES_O2STANDARD * 2
+	air_contents.nitrogen = MOLES_N2STANDARD
+	air_contents.temperature = T20C
+
+/obj/machinery/grey_autocloner/Destroy() //These will only be not null if deleted with someone in it
+	occupant = null
+	clonemind = null
+	return ..()
+
+/obj/machinery/grey_autocloner/proc/growclone(datum/dna2/record/R)
+	if(attempting || stat & (NOPOWER|BROKEN))
+		return 0
+	clonemind = locate(R.mind)
+	if(!istype(clonemind))	//not a mind
+		return 0
+	if(clonemind.current && clonemind.current.stat != DEAD)	//mind is associated with a non-dead body
+		return 0
+	if(clonemind.active)	//somebody is using that mind
+		if(ckey(clonemind.key) != R.ckey)
+			return 0
+		if(clonemind.suicided) // Nah, if you are being dragged to the borg factory, that's on you
+			return 0
+	else
+		// get_ghost() will fail if they're unable to reenter their body
+		var/mob/dead/observer/G = clonemind.get_ghost()
+		if(!G)
+			return 0
+
+	attempting = TRUE //One at a time!!
+
+	if(!R.dna)
+		R.dna = new /datum/dna()
+
+	var/mob/living/carbon/human/H = new /mob/living/carbon/human(src)
+	H.set_species(/datum/species/grey) //This is a grey cloner after all. Funnier this way tbh
+	occupant = H
+
+	if(!R.dna.real_name)	//to prevent null names
+		R.dna.real_name = H.real_name
+	else
+		H.real_name = R.dna.real_name
+
+	H.dna = R.dna.Clone()
+
+	for(var/datum/language/L in R.languages)
+		H.add_language(L.name)
+
+	domutcheck(H, MUTCHK_FORCED) //Ensures species that get powers by the species proc handle_dna keep them
+
+	H.dna.UpdateSE()
+	H.dna.UpdateUI()
+
+	H.sync_organ_dna(1) // It's literally a fresh body as you can get, so all organs properly belong to it
+	H.UpdateAppearance()
+
+	H.Paralyse(8 SECONDS)
+
+	clonemind.transfer_to(H) //INSTANTLY INTO THE CLONE
+	H.ckey = R.ckey
+	update_clone_antag(H) //Since the body's got the mind, update their antag stuff right now. Otherwise, wait until they get kicked out (as per the CLONER_MATURE_CLONE business) to do it.
+	var/message
+	message += "<b>Consciousness slowly creeps over you as your body regenerates.</b><br>"
+	message += "<i>So this is what cloning feels like?</i>"
+	to_chat(H, "<span class='notice'>[message]</span>")
+
+	update_icon()
+
+	H.suiciding = FALSE
+	attempting = FALSE
+	addtimer(CALLBACK(src, PROC_REF(finish_clone)), 1 MINUTES)
+	return 1
```

Comment:
consider not making them suffer the pains of being stuck in the machine for a minute
```suggestion
	return TRUE
```
Also is this isn't even being compared so you may want to just remove the returns

Reply:
Nope. No meta knowledge after death, and the fun apprehention if someone is coming while you are being revived with no way to be seen